app/src/main/java/de/danoeh/antennapod/adapter/DefaultActionButtonCallback.java

```diff
@@ -10,22 +12,31 @@
 import de.danoeh.antennapod.core.dialog.DownloadRequestErrorDialogCreator;
 import de.danoeh.antennapod.core.feed.FeedItem;
 import de.danoeh.antennapod.core.feed.FeedMedia;
-import de.danoeh.antennapod.core.service.playback.PlaybackService;
 import de.danoeh.antennapod.core.gpoddernet.model.GpodnetEpisodeAction;
 import de.danoeh.antennapod.core.preferences.GpodnetPreferences;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.service.playback.PlaybackService;
 import de.danoeh.antennapod.core.storage.DBTasks;
 import de.danoeh.antennapod.core.storage.DBWriter;
 import de.danoeh.antennapod.core.storage.DownloadRequestException;
 import de.danoeh.antennapod.core.storage.DownloadRequester;
+import de.danoeh.antennapod.core.util.NetworkUtils;
 
 /**
  * Default implementation of an ActionButtonCallback
  */
 public class DefaultActionButtonCallback implements ActionButtonCallback {
+
     private static final String TAG = "DefaultActionButtonCallback";
 
     private final Context context;
 
+    private final int TEN_MINUTES_IN_MILLIS = 60 * 1000 * 10;
+
+    // remember timestamp when user allowed downloading via mobile connection
+    private static long allowMobileDownloadsTimestamp;
```

Comment:
Do these need to be different? It never makes sense for them to both be set at the same time.

I can see it making the code more readable.

Should they be initialized to some value?